test-framework/sudo-compliance-tests/src/sudoers/run_as.rs

```diff
@@ -360,13 +363,37 @@ fn when_no_run_as_spec_then_a_target_group_may_be_specified() -> Result<()> {
     let output = Command::new("sudo")
         .args(["-u", "root", "-g", GROUPNAME, "groups"])
         .as_user(USERNAME)
+        .output(&env)?;
+
+    assert!(!output.status().success());
+    assert_eq!(Some(1), output.status().code());
+
+    let diagnostic = if sudo_test::is_original_sudo() {
+        format!("user {USERNAME} is not allowed to execute '/usr/bin/true' as root:{GROUPNAME}")
+    } else {
+        format!("I'm sorry {USERNAME}. I'm afraid I can't do that")
+    };
+    assert_contains!(output.stderr(), diagnostic);
+
+    Ok(())
+}
+
+#[test]
+fn when_no_run_as_spec_then_a_group_that_root_is_in_may_be_specified() -> Result<()> {
+    let env = Env("ALL ALL = NOPASSWD: ALL")
+        .user(User(USERNAME))
+        .group(GROUPNAME)
+        .build()?;
+
+    let output = Command::new("sudo")
+        .args(["-u", "root", "-g", "root", "groups"])
```

Comment:
I think it would be worthwhile to have another test case using a different group since the group root is the primary of the user root -- that is `sudo -u root groups` also prints `root`.

Reply:
Agreed, but a quick scan let me to believe that that requires a change to `sudo-test` (to be able to add root to a secondary group)?